src/routes/excalidraw.ts

```diff
@@ -0,0 +1,161 @@
+import { Router, Request, Response } from 'express';
+import { MinioService } from '../services/minioService';
+import { upload } from '../utils/multer';
+import { verifyFirebaseToken, skipAuthForDev, AuthRequest } from '../middleware/excalidrawAuth';
+
+const router = Router();
+const minioService = new MinioService();
+
+const isDev = process.env.NODE_ENV === 'development';
+const authMiddleware = isDev ? skipAuthForDev : verifyFirebaseToken;
+
+interface SceneDocument {
+  sceneVersion: number;
+  ciphertext: string; // base64 encoded
+  iv: string; // base64 encoded
+}
+
+// Save scene to MinIO
+router.post('/scenes', authMiddleware, async (req: AuthRequest, res: Response) => {
+  try {
+    const user = req.user;
+    console.log('Saving scene for user:', user?.email || user?.user_id);
+    
+    const { roomId, sceneVersion, ciphertext, iv }: any = req.body;
+
+    if (!roomId || !sceneVersion || !ciphertext || !iv) {
+      return res.status(400).json({ 
+        error: 'Missing required fields: roomId, sceneVersion, ciphertext, iv' 
+      });
+    }
+
+    const sceneDocument: SceneDocument = {
+      sceneVersion,
+      ciphertext,
+      iv,
+    };
+
+    const objectName = `scenes/${roomId}.json`;
+    const buffer = Buffer.from(JSON.stringify(sceneDocument), 'utf-8');
+    
+    await minioService.saveFile(objectName, buffer, 'application/json');
+
+    // console.log(`Saved scene: ${roomId}`);
+    res.json({ success: true, roomId, sceneVersion });
+  } catch (error: any) {
+    console.error('Error saving scene:', error);
+    res.status(500).json({ error: 'Failed to save scene', details: error.message });
+  }
+});
+
+// Load scene from MinIO
+router.get('/scenes/:roomId', async (req: Request, res: Response) => {
+  try {
+    const { roomId } = req.params;
+
+    if (!roomId) {
+      return res.status(400).json({ error: 'Room ID is required' });
+    }
+
+    const objectName = `scenes/${roomId}.json`;
+    const buffer = await minioService.getFile(objectName);
+
+    if (!buffer) {
+      return res.status(404).json({ error: 'Scene not found' });
+    }
+
+    const sceneDocument: SceneDocument = JSON.parse(buffer.toString('utf-8'));
+
+    res.json(sceneDocument);
+  } catch (error: any) {
+    console.error('Error loading scene:', error);
+    res.status(500).json({ error: 'Failed to load scene', details: error.message });
+  }
+});
+
+// Saving multiple files to MinIO using Multer
+router.post('/files/upload', authMiddleware, upload.array('files'), async (req: AuthRequest, res: Response) => {
```

Comment:
This server already handles file uploads and downloads with a different URL style.

Rather than adding new routes which do basically the same, modify excalidraw to send the requests to the Jitsi routes. This way we have a consistent API and we get to reuse the auth, for example.